core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -164,7 +164,15 @@ private[spark] class SparkSubmit extends Logging {
             }
         }
       } else {
-        runMain(childArgs, childClasspath, sparkConf, childMainClass, args.verbose)
+        if (sparkConf.getOption("spark.kubernetes.kerberos.proxyUser").isDefined) {
```

Comment:
Can we avoid hard-coding config name?

Reply:
@ifilonenko have you implemented proxy user support as well?

Reply:
@skonto this will be in a followup PR